pkg/auth/handler/gear/templates_test.go

```diff
@@ -0,0 +1,50 @@
+package gear
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	. "github.com/skygeario/skygear-server/pkg/core/skytest"
+	"github.com/skygeario/skygear-server/pkg/core/template"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestTemplatesHandler(t *testing.T) {
+	Convey("TemplatesHandler", t, func() {
+		Convey("should return information on gear templates", func() {
+			engine := template.NewEngine(template.NewEngineOptions{})
+			engine.Register(template.Spec{
+				Type:    "email.txt",
+				Default: "This is a test email.",
+			})
+			engine.Register(template.Spec{
+				Type:    "email.html",
+				Default: "",
+				IsHTML:  true,
+			})
+
+			var handler = TemplatesHandler{
+				TemplateEngine: engine,
+			}
+
+			r, _ := http.NewRequest("GET", "", nil)
+			rw := httptest.NewRecorder()
+			handler.ServeHTTP(rw, r)
+
+			So(rw.Body.Bytes(), ShouldEqualJSON, `[
```

Comment:
Not sure if [this](https://stackoverflow.com/questions/3503102/what-are-top-level-json-arrays-and-why-are-they-a-security-risk) is still valid?

Reply:
This is not sensitive information, so there's no need to be concerned about it.